Angular 18 + lazy loading + new config + new metadata update service + consolidation

chuanlin2018 opened the pull request:
This PR is for review only. Should not be merged into integration.

This branch has following changes:
1. Upgraded to Angular 18
2. Applied lazy loading to all components
3. Merged new config and metadata update service
4. Consolidation: separated public and midas.

Use node version: v20.16.0

Oar-lib-angular branch: angular-18-config (There are some changes in my local that I couldn't see the details so I didn't check in. May need to manually switch to this branch).

All installation and build steps are the same as before.
To test public side locally, after run "npm run start-pdrlps", you should be able to use http://localhost:4201/ to browser. It's currently configured to use data.nist.gov as backend.

http://localhost:4201/od/id/5CF421B31BE80304E0531A57068197E41874
http://localhost:4201/od/id/pdr0-0001
http://localhost:4201/od/id/ark:/88434/mds2-2294

Testing midas side locally is a little tricky.
1. First need to enable the fake backend in app.module.ts. 
2. It's configured to use local docker as backend (localhost) so current version docker need be run in local. 
3. In _helpers/fakebackendinterceptor.ts, need to provide current token in line 209. The token should be available here:
       https://localhost/sso/auth/_tokeninfo
Then you should be able to test http://localhost:4202/od/id/mds3:0001?editenabled=true (assume mds3:0001 is available).

Let me know if you have any question.

RayPlante reviewed (changes requested):
Ran into a build issue:  

> Cannot read file /dev/oar-pdr-angular/oar-lps/libs/oarlps/src/lib/landing/visithome/visithome-pub/Visithome-pub.component.scss.

This appears to be due to a typo in `lib/landing/visithome/visithome-pub/visithome-pub.component.ts`: 
```javascript
    styleUrls: ['./Visithome-pub.component.scss', '../../landing.component.scss'],
```
The file is actually called `visithome-pub.component.scss`.  The same applies to `visithome-midas.component.ts`.

chuanlin2018 commented:
Fixed the capital V. Also fixed doi link issue and landing page display timeout issue. The current oar-pdr could have the same display timeout issue (wait for 15s to display).

chuanlin2018 commented:
I believe my compiler was set to case insensitive so it built in my local without problem.

RayPlante reviewed (commented):
Build fixes have been addressed and I'm able to run the editable version.  I've tested a bunch of things everything is basically looking good.  I'm not going to test absolutely everything; I think we'll handle any remaining issues as future bugs.  We'll discuss merging at our next design meeting.

For now, I'll just mention one thing about rendering `accessURL`s (e.g. links to software repositories).  Currently, the link description is not getting rendered.  In the legacy public page, if there is a description, an arrow-triangle is displayed after the link; opening it will display the description below the link.  Let's put that into this codebase.

RayPlante commented:
superceded by #63; closing without merging.
